### PR TITLE
Add ability to update the service instance

### LINKED ISSUE #19: Add ability to update service instance

### PR DESCRIPTION
**Pull request by LiliC**
This PR adds ability to update the service instance. Let me know in case I missed something?

Closes #19.

### CODE REVIEW

File: pkg/rest/apisurface.go [hunk 1]
```diff
@@ -265,3 +265,45 @@ func writeError(w http.ResponseWriter, err error) {
...
+
+	writeResponse(w, status, response)
+}
+
+func unpackUpdateRequest(r *http.Request) (*osb.UpdateInstanceRequest, error) {
+	osbRequest := &osb.UpdateInstanceRequest{}
+
+	vars := mux.Vars(r)
+	osbRequest.ServiceID = vars[serviceIDVarKey]
+	planID := vars[planIDVarKey]
```
Comment (LiliC) [inline pkg/rest/apisurface.go]: This has to be done as the osb-client expects a string pointer. https://github.com/pmorie/go-open-service-broker-client/blob/master/v2/update_instance.go#L12

I am assuming that was done to indicate the PlanID is not mandatory as per the spec?

**Comment by LiliC** [comment]
Forgot to update the test I created in the first place. :woman_facepalming: Should be fixed now! :)

File: pkg/rest/apisurface.go (hunk 1 above)
Comment (pmorie) [reply pkg/rest/apisurface.go]: That's correct - the planID is sent, according to the spec, only when the user explicitly requests that the plan should change.  See: https://github.com/openservicebrokerapi/servicebroker/blob/master/spec.md#body-4

So, you won't always get it.  I'm thinking that if it isn't sent to you, shouldn't get a pointer to an empty string.  Let's check the value of the var and if it's empty, avoid setting the pointer.  WDYT?

**Comment by LiliC** [comment]
@pmorie Done! PTALA.

**Pull request merged**